### PR TITLE
 lazy compute input ids

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Make input ids computation to lazy, not required all the time

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]

File: engines/python/setup/djl_python/input_parser.py [hunk 1]
```diff
@@ -148,10 +148,6 @@ def parse_text_inputs_params(request_input: TextInput, input_item: Input,
 
     request_input.input_text = inputs
     request_input.parameters = param
-    # assigns input_ids
-    # TODO: for dynamic batching, or HF pipeline, tokenizer is applied differently.
-    if kwargs.get("tokenizer") and kwargs.get("is_rolling_batch"):
-        request_input.input_ids = tokenizer.encode(request_input.input_text)
```
Comment (davidthomas426) [inline engines/python/setup/djl_python/input_parser.py:154]: Do we need the tokenizer at all anymore in this function?

**Review by davidthomas426 — APPROVED** [review]

File: engines/python/setup/djl_python/input_parser.py (hunk 1 above)
Comment (siddvenk) [reply engines/python/setup/djl_python/input_parser.py:154]: It is needed for chat_completions and bedrock use-cases to apply the chat-template.

File: engines/python/setup/djl_python/rolling_batch/rolling_batch_vllm_utils.py [hunk 2]
```diff
@@ -43,15 +43,15 @@ def update_request_cache_with_output(request_cache: OrderedDict,
...
+    # sets prompt token details if not set
     if vllm_request_output.prompt_logprobs and not request_output.prompt_tokens_details:
         # TODO: Temp check adding the check fo T5.
         if isinstance(vllm_request_output.prompt_token_ids, list):
             for index, prompt_token_id in enumerate(
                     vllm_request_output.prompt_token_ids):
                 prompt_token = Token(
                     id=prompt_token_id,
-                    text=tokenizer.decode([prompt_token_id]),
+                    text=tokenizer.convert_ids_to_tokens(prompt_token_id),
```
Comment (siddvenk) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch_vllm_utils.py:54]: We can choose to leave this as None, but this method is just a dict lookup

**Comment by siddvenk** [comment]
I ran some benchmarks to understand the impact of these changes:

For all tests, I used the command:
```
TOKENIZER=llama3-8b ./awscurl -c 64 -N 1000 -X POST http://127.0.0.1:8080/invocations \
  --connect-timeout 120  \
  --duration 300   \
 --delay "rand(0, 10000)"  \
 -H "Content-type: application/json"   \
 --dataset openorca_djlserving.jsonl  \
 -t -P -o response.txt
```

v0.28.0 container:
```
{
  "tokenizer": "llama3-8b",
  "totalTimeMills": 306441.11,
  "totalRequests": 1038,
  "failedRequests": 0,
  "errorRate": 0.0,
  "concurrentClients": 64,
  "tps": 3.55,
  "tokenThroughput": 632.39,
  "totalTokens": 185078,
  "tokenPerRequest": 178,
  "averageLatency": 18044.76,
  "p50Latency": 23678.73,
  "p90Latency": 24960.39,
  "p99Latency": 26134.46,
  "timeToFirstByte": 5117.05,
  "p50TimeToFirstByte": 4292.293989,
  "p90TimeToFirstByte": 10004.263995,
  "p99TimeToFirstByte": 13723.521713
}
```

current lmi-nightly without changes in this PR:

```
{
  "tokenizer": "llama3-8b",
  "totalTimeMills": 309870.87,
  "totalRequests": 995,
  "failedRequests": 0,
  "errorRate": 0.0,
  "concurrentClients": 64,
  "tps": 3.37,
  "tokenThroughput": 601.01,
  "totalTokens": 177572,
  "tokenPerRequest": 178,
  "averageLatency": 19004.11,
  "p50Latency": 24581.14,
  "p90Latency": 25958.34,
  "p99Latency": 27906.81,
  "timeToFirstByte": 5733.94,
  "p50TimeToFirstByte": 5107.926974,
  "p90TimeToFirstByte": 10124.48556,
  "p99TimeToFirstByte": 14215.705665
}
```

current lmi + changes in this PR:

```
{
  "tokenizer": "llama3-8b",
  "totalTimeMills": 308637.27,
  "totalRequests": 1037,
  "failedRequests": 0,
  "errorRate": 0.0,
  "concurrentClients": 64,
  "tps": 3.52,
  "tokenThroughput": 633.63,
  "totalTokens": 186596,
  "tokenPerRequest": 179,
  "averageLatency": 18174.85,
  "p50Latency": 23589.39,
  "p90Latency": 24972.48,
  "p99Latency": 26618.66,
  "timeToFirstByte": 5423.92,
  "p50TimeToFirstByte": 4612.326048,
  "p90TimeToFirstByte": 9597.682596,
  "p99TimeToFirstByte": 13975.139746
}
```

**Review by lanking520 — COMMENTED** [review]
LGTM, thank you @siddvenk

**Pull request merged**